src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -238,189 +230,80 @@ export const useSwapper = () => {
     })
-
-    const result = await swapper.executeTrade({ trade: quote, wallet })
-    return result
+    return swapper.executeTrade({ trade, wallet })
   }
 
-  const getQuoteFromSwapper = async <C extends ChainTypes>({
-    amount,
-    sellAsset,
-    buyAsset,
-    action,
-    onFinish,
-  }: GetQuoteFromSwapper<C>) => {
-    if (debounceObj?.cancel) debounceObj.cancel()
-    clearErrors()
-    const quoteDebounce = debounce(async () => {
-      if (isComponentMounted.current) {
-        try {
-          const swapper = await swapperManager.getBestSwapper({
-            buyAssetId: buyAsset.assetId,
-            sellAssetId: sellAsset.assetId,
-          })
-          let convertedAmount =
-            action === TradeActions.BUY ? { buyAmount: amount } : { sellAmount: amount }
-
-          if (action === TradeActions.FIAT) {
-            const rate = bn(
-              await swapper.getUsdRate({
-                symbol: sellAsset.symbol,
-                tokenId: sellAsset.tokenId,
-              }),
-            )
-            convertedAmount = {
-              sellAmount: rate.gt(0)
-                ? toBaseUnit(bn(amount).div(rate).toString(), sellAsset.precision).toString()
-                : '0',
-            }
-          }
-          const quoteInput = {
-            sellAsset,
-            buyAsset,
-            ...convertedAmount,
-          }
-
-          const { trade } = getValues()
-          let minMax = trade
-
-          if (
-            !quote ||
-            (quote?.sellAsset?.symbol !== sellAsset.symbol &&
-              quote?.buyAsset?.symbol !== buyAsset.symbol)
-          ) {
-            minMax = await swapper.getMinMax(quoteInput)
-            const minMaxTrade = { ...minMax, ...trade }
-            minMax && setValue('trade', minMaxTrade)
-          }
-
-          const newQuote = await swapper.getTradeQuote({
-            ...quoteInput,
-            ...minMax,
-          })
-          if (!(newQuote && newQuote.success)) throw newQuote
-
-          const sellAssetUsdRate = bnOrZero(
-            await swapper.getUsdRate({
-              symbol: sellAsset.symbol,
-              tokenId: sellAsset.tokenId,
-            }),
-          )
-          const newQuoteRate = bnOrZero(newQuote.rate)
-          const buyAssetUsdRate = newQuoteRate.gt(0)
-            ? sellAssetUsdRate.div(newQuoteRate).toString()
-            : bnOrZero(
-                await swapper.getUsdRate({
-                  symbol: buyAsset.symbol,
-                  tokenId: buyAsset.tokenId,
-                }),
-              )
-          // TODO: Implement fiatPerUsd here
-          setFees(newQuote, sellAsset)
-          setValue('quote', newQuote)
-          setValue('sellAsset.fiatRate', sellAssetUsdRate.toString())
-          setValue('buyAsset.fiatRate', buyAssetUsdRate.toString())
-          if (action) onFinish(newQuote)
-        } catch (e) {
-          console.error(e)
-        }
+  const updateQuoteDebounced = useRef(
```

Comment:
What's the rationale for using `useRef` here?

Reply:
In order for lodash debounce to work correctly here it needs to use the single reference is my understnaiong. 

Other people online recommend doing it like this but if theres a better way im down

Reply:
Wouldn't `useCallback` keep the referential stability without having to deal with refs?